tests/automated_validation/data_transformation/test_formatting.py

```diff
@@ -0,0 +1,110 @@
+import pandas as pd
+
+from vivarium_testing_utils.automated_validation.data_transformation.formatting import (
+    PersonTime,
+    TransitionCounts,
+    drop_redundant_index,
+)
+
+
+def test_drop_redundant_index() -> None:
+    """Test drop_redundant_index function."""
+    # Create a mock dataset
+    dataset = pd.DataFrame(
+        {
+            "value": [10, 20, 30, 40],
+        },
+        index=pd.MultiIndex.from_tuples(
+            [
+                ("redundant_column", "heterogeneous"),
+                ("redundant_column", "values"),
+                ("redundant_column", "in_this"),
+                ("redundant_column", "column"),
+            ],
+            names=["redundant_column", "interesting_column"],
+        ),
+    )
+    expected_dataframe = pd.DataFrame(
+        {
+            "value": [10, 20, 30, 40],
+        },
+        index=pd.Index(
+            ["heterogeneous", "values", "in_this", "column"],
+            name="interesting_column",
+        ),
+    )
+    # Call the function to drop the redundant index
+    formatted_dataset = drop_redundant_index(dataset, "redundant_column", "redundant_column")
+    assert formatted_dataset.equals(expected_dataframe)
+
+
+def test_transition_counts(transition_count_data: pd.DataFrame) -> None:
+    """Test TransitionCounts formatting."""
+    formatter = TransitionCounts("disease", "susceptible_to_disease", "disease")
+    assert formatter.type == "transition_count"
```

Comment:
I often like to assert that the len of attrs is whatever to catch if you ever change the class in the future. Would then force you to think if you need to improve the test.